Penteract/Assets/Scripts/BridgeDoorButton.cpp

```diff
@@ -0,0 +1,64 @@
+#include "BridgeDoorButton.h"
+
+#include "FactoryDoors.h"
+#include "FloorIsLava.h"
+#include "GameplaySystems.h"
+#include "GameObject.h"
+
+
+EXPOSE_MEMBERS(BridgeDoorButton) {
+	MEMBER(MemberType::GAME_OBJECT_UID, bridgeDoorUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, fireBridgeUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, fireArenaUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, doorObstacleUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, laserDoorObstacleUID)
+};
+
+GENERATE_BODY_IMPL(BridgeDoorButton);
+
+void BridgeDoorButton::Start() {
+	gameObject = &GetOwner();
+	GameObject* bridgeDoor = GameplaySystems::GetGameObject(bridgeDoorUID);
+	if (bridgeDoor) {
+		doorScript = GET_SCRIPT(bridgeDoor, FactoryDoors);
+	}
+	GameObject* fireBridge = GameplaySystems::GetGameObject(fireBridgeUID);
+	if (fireBridge) {
+		bridgeTilesScript = GET_SCRIPT(fireBridge, FloorIsLava);
+	}
+	GameObject* fireArena = GameplaySystems::GetGameObject(fireArenaUID);
+	if (fireArena) {
+		arenaTilesScript = GET_SCRIPT(fireArena, FloorIsLava);
+	}
+
+	doorObstacle = GameplaySystems::GetGameObject(doorObstacleUID);
+
+	laserDoorObstacle = GameplaySystems::GetGameObject(laserDoorObstacleUID);
+
+}
+
+void BridgeDoorButton::Update() {
+	if (laserDoorObstacle->GetComponent<ComponentMeshRenderer>()->HasDissolveAnimationFinished()) {
```

Comment:
if (laserDoorObstacle && laserDoorObstacle->GetComponent...), otherwise, this will return a null pointer exception if scene isn't correctly set

Reply:
FIxed